libbeat/esleg/eslegclient/connection.go

```diff
@@ -452,7 +452,9 @@ func (conn *Connection) execHTTPRequest(req *http.Request) (int, []byte, error)
 	defer closing(resp.Body, conn.log)
 
 	status := resp.StatusCode
-	obj, err := ioutil.ReadAll(resp.Body)
+	obj := new(bytes.Buffer)
+	_, err = io.Copy(obj, resp.Body)
+	// obj, err := ioutil.ReadAll(resp.Body)
```

Comment:
You can remove this line.
```suggestion
```

Reply:
nits:
Would be nice to benchmark.
Also, could possibly check the ContentLength if available and preallocate the buffer to the size.
```
if resp.ContentLength > 0 {
    buf = bytes.NewBuffer(make([]byte, 0, resp.ContentLength))
```

Reply:
@aleksmaus see https://github.com/elastic/beats/issues/36151#issuecomment-1931696767